src/Propel/Generator/Model/Table.php

```diff
@@ -784,6 +784,25 @@ public function setupReferrers($throwErrors = false)
                 }
             }
 
+            // check for incomplete foreign key references when foreign table
+            // has a composite primary key
+            if ($foreignTable->hasCompositePrimaryKey()) {
+                // get composite foreign key's keys
+                $foreignPrimaryKeys = $foreignTable->getPrimaryKey();
+                // check all keys are referenced in foreign key
+                foreach ($foreignPrimaryKeys as $foreignPrimaryKey) {
+                    if (!$foreignPrimaryKey->hasReferrer($foreignKey) && $throwErrors) {
```

Comment:
Do you think we could cover this use case with this schema ?

`<database name="test" defaultIdMethod="native">
    <table name="event">
        <column name="id" type="integer" required="true" primaryKey="true" autoIncrement="true" />
        <column name="name" type="varchar" size="50" required="true" />
        <column name="organiser_id" type="integer" required="true" />
        <!-- This FK is incomplete -->
        <foreign-key foreignTable="organiser">
            <reference local="organiser_id" foreign="id" />
        </foreign-key>
    </table>

    <table name="organiser">
        <column name="id" type="integer" required="true" primaryKey="true" autoIncrement="true" />
        <column name="code" type="integer" required="true"/>
        <column name="name" type="varchar" size="50" required="true" />
        <!-- Has unique -->
        <unique name="u_id_code">
          <unique-column name="id" />
          <unique-column name="code" />
        </unique>
    </table>
</database>`